### PR TITLE
Add some System.DirectoryServices tests that don't require a server

### PR DESCRIPTION
**Pull request by hughbe**
I quickly realised that to properly test the library you need a server, but here's what I managed so far without one

### CODE REVIEW

**Comment by hughbe** [comment]
@danmosemsft funny looking Windows Nano failure here? It's expected though I think...

```
Assert.Throws() Failure\r\nExpected: typeof(System.Runtime.InteropServices.COMException)\r\nActual:   typeof(System.DllNotFoundException): Unable to load DLL 'activeds.dll': The specified module could not be found. (Exception from HRESULT: 0x8007007E)

```

**Comment by danmosemsft** [comment]
@hughbe yes, you will have to disable the tests for nano as apparently those components aren't available.

We probably need to find a way to do this more conveniently..

**Comment by hughbe** [comment]
CI stuck on
```
08:11:21 Body did not finish within grace period; terminating with extreme prejudice
[Pipeline] }
08:11:21 ERROR: Execution failed
08:11:21 org.jenkinsci.plugins.workflow.steps.FlowInterruptedException
08:11:21 	at org.jenkinsci.plugins.workflow.steps.TimeoutStepExecution.cancel(TimeoutStepExecution.java:103)
08:11:21 	at org.jenkinsci.plugins.workflow.steps.TimeoutStepExecution.access$100(TimeoutStepExecution.java:24)
08:11:21 	at org.jenkinsci.plugins.workflow.steps.TimeoutStepExecution$1.run(TimeoutStepExecution.java:83)
08:11:21 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
08:11:21 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
08:11:21 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
08:11:21 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
08:11:21 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
08:11:21 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
08:11:21 	at java.lang.Thread.run(Thread.java:748)
08:11:21 Retrying
[Pipeline] {
[Pipeline] checkout
08:11:22 Cloning the remote Git repository
08:11:22 Cloning repository https://github.com/dotnet/corefx
```

**Comment by hughbe** [comment]
Test Portable Windows x64 Debug Build

**Comment by danmosemsft** [comment]
@hughbe when there are infrastructure issues, the entity to summon is @dotnet/dnceng ... they'd need a link to the job..

**Comment by hughbe** [comment]
And I just deleted it by retriggering the tests right, sorry

**Comment by hughbe** [comment]
Updated, found it in my history: https://ci3.dot.net/job/dotnet_corefx/job/master/job/portable-windows-Config_Debug+OuterLoop_false_prtest/1076/console
[consoleText.txt](https://github.com/dotnet/corefx/files/1083587/consoleText.txt)

**Comment by danmosemsft** [comment]
@dotnet/dnceng please note this:
```
06:09:15 Cloning repository https://github.com/dotnet/corefx
08:10:21 Cancelling nested steps due to timeout
[Pipeline] }
08:10:21 ERROR: Execution failed
08:10:21 java.lang.InterruptedException
08:10:21 	at java.lang.Object.wait(Native Method)
08:10:21 	at hudson.remoting.Request.call(Request.java:147)
08:10:21 	at hudson.remoting.Channel.call(Channel.java:829)
08:10:21 	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler.execute(RemoteGitImpl.java:146)
08:10:21 	at sun.reflect.GeneratedMethodAccessor464.invoke(Unknown Source)
08:10:21 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
08:10:21 	at java.lang.reflect.Method.invoke(Method.java:498)
08:10:21 	at org.jenkinsci.plugins.gitclient.RemoteGitImpl$CommandInvocationHandler.invoke(RemoteGitImpl.java:132)
08:10:21 	at com.sun.proxy.$Proxy176.execute(Unknown Source)
08:10:21 	at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1067)
08:10:21 	at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1107)
08:10:21 	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep.checkout(SCMStep.java:109)
08:10:21 	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:83)
08:10:21 	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:73)
08:10:21 	at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1$1.call(AbstractSynchronousNonBlockingStepExecution.java:47)
08:10:21 	at hudson.security.ACL.impersonate(ACL.java:260)
08:10:21 	at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1.run(AbstractSynchronousNonBlockingStepExecution.java:44)
08:10:21 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
08:10:21 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
08:10:21 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
08:10:21 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
08:10:21 	at java.lang.Thread.run(Thread.java:748)
08:10:21 Retrying
[Pipeline] {
[Pipeline] checkout
08:10:22 Cloning the remote Git repository
08:10:22 Cloning repository https://github.com/dotnet/corefx
08:11:21 Body did not finish within grace period; terminating with extreme prejudice
[Pipeline] }
08:11:21 ERROR: Execution failed
08:11:21 org.jenkinsci.plugins.workflow.steps.FlowInterruptedException
08:11:21 	at org.jenkinsci.plugins.workflow.steps.TimeoutStepExecution.cancel(TimeoutStepExecution.java:103)
08:11:21 	at org.jenkinsci.plugins.workflow.steps.TimeoutStepExecution.access$100(TimeoutStepExecution.java:24)
08:11:21 	at org.jenkinsci.plugins.workflow.steps.TimeoutStepExecution$1.run(TimeoutStepExecution.java:83)
08:11:21 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
08:11:21 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
08:11:21 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
08:11:21 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
08:11:21 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
08:11:21 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
08:11:21 	at java.lang.Thread.run(Thread.java:748)
08:11:21 Retrying
[Pipeline] {
[Pipeline] checkout
08:11:22 Cloning the remote Git repository
08:11:22 Cloning repository https://github.com/dotnet/corefx
```

**Comment by karelz** [comment]
@stephentoub can you please help us with code review?

**Comment by hughbe** [comment]
Test Linux x64 Release Build 
Test NETFX x86 Release Build

File: src/System.DirectoryServices/tests/System/DirectoryServices/DirectoryEntryTests.cs [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+            Assert.Throws<ObjectDisposedException>(() => entry.SchemaEntry);
+        }
+
+        [Fact]
+        public void Dispose_InvokeMethods_ThrowsObjectDisposedException()
+        {
+            var entry = new DirectoryEntry("path");
+            entry.Dispose();
+
+            //Assert.Throws<ObjectDisposedException>(() => entry.CopyTo(new DirectoryEntry()));
```
Comment (stephentoub) [inline src/System.DirectoryServices/tests/System/DirectoryServices/DirectoryEntryTests.cs]: Why is this commented out?

**Pull request merged**